common/flatpak-installation.c

```diff
@@ -1113,9 +1116,15 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
        * This makes sure that the ref (maybe an app or runtime) remains in usable
        * state and fixes itself through an update.
        */
-      state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, error);
+      state = g_hash_table_lookup (remote_states, remote_name);
```

Comment:
This here assigns the hash lookup to an autoptr, which will then be freed.

I think the proper ownership here is that all refs are owned by the hashtable, so make the state local var not autoptr, and drop the ref() below so that the created ref is moved to the hashtable.

Reply:
Thanks. Addressed.